onnxruntime/core/providers/trt/trt_execution_provider.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <ctime>
+#include "core/framework/op_kernel.h"
+#include "NvInfer.h"
+#include "NvOnnxParser.h"
+
+namespace onnxruntime {
+
+static const int kBatchSize = 1;
+static const int max_batch_size = 1;
+static const int max_workspace_size = 1 << 30;
+
+#define CHECK(status)                       \
+  do {                                      \
+    auto ret = (status);                    \
+    if (ret != 0) {                         \
+      std::cout << "Cuda failure: " << ret; \
+      abort();                              \
+    }                                       \
+  } while (0)
+
+struct InferDeleter {
+  template <typename T>
+  void operator()(T* obj) const {
+    if (obj) {
+      obj->destroy();
+    }
+  }
+};
+
+template <typename T>
+using unique_pointer = std::unique_ptr<T, InferDeleter>;
+
+class TRTLogger : public nvinfer1::ILogger {
+  nvinfer1::ILogger::Severity verbosity_;
+  std::ostream* ostream_;
+
+ public:
+  TRTLogger(Severity verbosity = Severity::kWARNING,
+            std::ostream& ostream = std::cout)
+      : verbosity_(verbosity), ostream_(&ostream) {}
+  void log(Severity severity, const char* msg) override {
+    ORT_UNUSED_PARAMETER(severity);
+    ORT_UNUSED_PARAMETER(msg);
+  }
```

Comment:
This needs to use our logger for the output not an ostream. Otherwise we have log messages going to different places, which is not good as a user experience.